chips/swerv/src/eh1_timer.rs

```diff
@@ -0,0 +1,213 @@
+        // We have no way to know when this happens
+    }
+
+    fn start(&self) -> ReturnCode {
+        match self.number {
+            TimerNumber::ZERO => self.mitctl0.modify(MITCTL::ENABLE::SET),
+            TimerNumber::ONE => self.mitctl1.modify(MITCTL::ENABLE::SET),
+        };
+
+        ReturnCode::SUCCESS
+    }
+
+    fn stop(&self) -> ReturnCode {
+        match self.number {
+            TimerNumber::ZERO => self.mitctl0.modify(MITCTL::ENABLE::CLEAR),
+            TimerNumber::ONE => self.mitctl1.modify(MITCTL::ENABLE::CLEAR),
+        };
+
+        ReturnCode::SUCCESS
+    }
+
+    fn reset(&self) -> ReturnCode {
+        // A counter is only cleared when it is equal or greater then
+        // mitb.
+        ReturnCode::FAIL
+    }
+
+    fn is_running(&self) -> bool {
+        match self.number {
+            TimerNumber::ZERO => self.mitctl0.is_set(MITCTL::ENABLE),
+            TimerNumber::ONE => self.mitctl1.is_set(MITCTL::ENABLE),
+        }
+    }
+}
+
+impl<'a> Alarm<'a> for Timer<'a> {
+    fn set_alarm_client(&self, client: &'a dyn time::AlarmClient) {
+        self.alarm_client.set(client);
+    }
+
+    fn set_alarm(&self, reference: Self::Ticks, dt: Self::Ticks) {
+        // Start the counter
+        if !self.is_running() {
+            Counter::start(self);
+        }
+
+        let now = self.now();
+        let mut expire = reference.wrapping_add(dt);
+
+        if !now.within_range(reference, expire) {
+            expire = now;
+        }
+
+        let mut val = expire.into_usize();
+
+        // 0xFFFF_FFFF is reserved to indicate disabled, don't set that value
+        if val == 0xFFFF_FFFF {
+            val += 1;
+        }
+
+        match self.number {
+            TimerNumber::ZERO => self.mitb0.write(MITB::BOUND.val(val)),
+            TimerNumber::ONE => self.mitb1.write(MITB::BOUND.val(val)),
+        }
+    }
+
+    fn get_alarm(&self) -> Self::Ticks {
+        match self.number {
+            TimerNumber::ZERO => Self::Ticks::from(self.mitb0.read(MITB::BOUND) as u32),
+            TimerNumber::ONE => Self::Ticks::from(self.mitb1.read(MITB::BOUND) as u32),
+        }
+    }
+
+    fn disarm(&self) -> ReturnCode {
+        match self.number {
+            TimerNumber::ZERO => self.mitb0.write(MITB::BOUND.val(0xFFFF_FFFF)),
+            TimerNumber::ONE => self.mitb1.write(MITB::BOUND.val(0xFFFF_FFFF)),
+        };
+
+        ReturnCode::SUCCESS
+    }
+
+    fn is_armed(&self) -> bool {
+        match self.number {
+            TimerNumber::ZERO => self.mitb0.read(MITB::BOUND) != 0xFFFF_FFFF,
+            TimerNumber::ONE => self.mitb1.read(MITB::BOUND) != 0xFFFF_FFFF,
+        }
+    }
+
+    fn minimum_dt(&self) -> Self::Ticks {
+        Self::Ticks::from(1 as u32)
+    }
+}
+
+impl kernel::SchedulerTimer for Timer<'_> {
+    fn start(&self, us: u32) {
+        let now = self.now();
+        let tics = Self::ticks_from_us(us);
```

Comment:
note that `ticks_from_us()` rounds down, so if us is less than 10000 this will set an alarm for `now`

Reply:
Resolved by the change to higher frequency